Objects/typeobject.c

```diff
@@ -5297,12 +5297,12 @@ get_base_by_token_recursive(PyTypeObject *type, void *token)
     Py_ssize_t n = PyTuple_GET_SIZE(bases);
     for (Py_ssize_t i = 0; i < n; i++) {
         PyTypeObject *base = _PyType_CAST(PyTuple_GET_ITEM(bases, i));
-        if (!_PyType_HasFeature(base, Py_TPFLAGS_HEAPTYPE)) {
-            continue;
-        }
-        if (((PyHeapTypeObject*)base)->ht_token == token) {
+        // The MRO version can slow down if this takes the same approach,
+        // which also affects the same conditions in the functions above.
+        if (PyType_GetSlot(base, Py_tp_token) == token) {
```

Comment:
I'd prefer reverting this change: in `typeobject.c`, use the type internals directly.

Or did this somehow improve performance?

Reply:
Do you mean the performance of this recursive function?  If so, I'll reconsider.

Reply:
I meant that as a question: did you do this change to improve performance?